website/source/docs/internals/telemetry.html.md

```diff
@@ -168,13 +161,28 @@ These metrics relate to supported authentication methods.
 | `vault.route.rollback.secret` | Time taken to perform a route rollback operation for the [K/V secret backend][kv-secrets-engine] | ms | summary |
 | `vault.route.rollback.sys` | Time taken to perform a route rollback operation for the system backend | ms | summary |
 
+## Merkle Tree and Write Ahead Log Metrics
+
+These metrics relate to internal operations on Merkle Trees and Write Ahead Logs (WAL)
+
+| Metric | Description | Unit | Type |
+| :----- | :---------- | :--- | :--- |
+| `vault.merkle_flushdirty` | Time taken to flush any dirty pages to cold storage | ms | summary |
+| `vault.merkle_savecheckpoint` | Time taken to save the checkpoint | ms | summary |
+| `vault.wal_deletewals` | Time taken to delete a Write Ahead Log (WAL) | ms | summary |
+| `vault.wal_gc_deleted` | Number of Write Ahead Logs (WAL) deleted during each garbage collection run | WAL | counter |
+| `vault.wal_gc_total` | Total Number of Write Ahead Logs (WAL) on disk | WAL | counter |
+| `vault.wal_loadWAL` | Time taken to load a Write Ahead Log (WAL) | ms | summary |
+| `vault.wal_persistwals` | Time taken to persist a Write Ahead Log (WAL) | ms | summary |
+| `vault.wal_flushready` | Time taken to flush a ready Write Ahead Log (WAL) to storage | ms | summary |
```

Comment:
There are a few merkle and WAL related metrics in the section below, do they belong up here now?

Reply:
I think the ones below are all replication-specific and already in the Replication section and have _replication_ as part of their name.

It seems these other metrics don't have any specific association or subsystem name and that is why I grouped them together like this, but if it doesn't make sense to or the section should be different then I will change it.

Reply:
Ah, i see what you mean. Yeah i think this is good 👍